packages/strapi-connector-mongoose/lib/buildQuery.js

```diff
@@ -178,16 +179,23 @@ const buildDeepQuery = ({ model, filters, search, populate }) => {
             )
             .populate(populate);
 
-          const stringIds = ids.map(id => id.toString());
-          const getIndexInIds = obj => stringIds.indexOf(obj._id.toString());
+          const idsMapper = ids
+            .map(id => id.toString())
+            .reduce((acc, id, idx) => ({ ...acc, [id]: idx }), {});
```

Comment:
You can remove the map here when using the id as a key it will be cast to a string

Reply:
Thanks for the tip!